sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java

```diff
@@ -213,17 +221,53 @@ public void serviceHandlerThrowsException() {
             .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
             .build());
     ReadableSpan span1 = createSampledEndedSpan(SPAN_NAME_1);
-    List<SpanData> exported = waitingSpanExporter.waitForExport(1);
+    List<SpanData> exported = waitingSpanExporter.waitForExport();
     assertThat(exported).containsExactly(span1.toSpanData());
-
+    waitingSpanExporter.reset();
     // Continue to export after the exception was received.
     ReadableSpan span2 = createSampledEndedSpan(SPAN_NAME_2);
-    exported = waitingSpanExporter.waitForExport(1);
+    exported = waitingSpanExporter.waitForExport();
     assertThat(exported).containsExactly(span2.toSpanData());
   }
 
+  @Test(timeout = 200)
+  public void exporterTimesOut() throws Exception {
+    final CountDownLatch interruptMarker = new CountDownLatch(1);
+    WaitingSpanExporter waitingSpanExporter =
+        new WaitingSpanExporter(1) {
+          @Override
+          public ResultCode export(List<SpanData> spans) {
+            ResultCode result = super.export(spans);
+            try {
+              // sleep longer than the configured timout of 100ms
+              Thread.sleep(1000);
+            } catch (InterruptedException e) {
+              interruptMarker.countDown();
+            }
+            return result;
+          }
+        };
+
+    tracerSdkFactory.addSpanProcessor(
+        BatchSpansProcessor.newBuilder(waitingSpanExporter)
+            .setExporterTimeoutMillis(100)
+            .setScheduleDelayMillis(1)
+            .setMaxQueueSize(1)
+            .build());
+
+    ReadableSpan span = createSampledEndedSpan(SPAN_NAME_1);
+    List<SpanData> exported = waitingSpanExporter.waitForExport();
+    assertThat(exported).containsExactly(span.toSpanData());
+
+    // since the interrupt happens outside the execution of the test method, we'll block to make
+    // sure that the thread was actually interrupted due to the timeout.
+    // The test itself will fail if the test timeout is exceeded due to this waiting.
+    interruptMarker.await();
```

Comment:
The timeout from the test can be moved here.

Reply:
very true! I didn't realize that the latch returned false if the timeout expired. Great idea!

Reply:
done